File: src/main/java/org/keynote/godtools/android/sync/BaseSyncTasks.java
package org.keynote.godtools.android.sync;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.WorkerThread;

import org.keynote.godtools.android.api.GodToolsApi;
import org.keynote.godtools.android.db.GodToolsDao;

import static android.content.ContentResolver.SYNC_EXTRAS_MANUAL;

@WorkerThread
abstract class BaseSyncTasks {
    private final Context mContext;
    final GodToolsApi mApi;
    final GodToolsDao mDao;

    BaseSyncTasks(@NonNull final Context context) {
        mContext = context;
        mApi = GodToolsApi.getInstance(mContext);
        mDao = GodToolsDao.getInstance(mContext);
    }

    static boolean isForced(@NonNull final Bundle extras) {
        return extras.getBoolean(SYNC_EXTRAS_MANUAL, false);
    }
}

File: library/sync/src/main/kotlin/org/cru/godtools/sync/task/BaseSyncTasks.kt
package org.cru.godtools.sync.task

import android.content.ContentResolver
import android.os.Bundle
import androidx.annotation.RestrictTo
import androidx.annotation.WorkerThread
import androidx.collection.LongSparseArray
import org.cru.godtools.model.Base

@WorkerThread
@RestrictTo(RestrictTo.Scope.LIBRARY)
abstract class BaseSyncTasks internal constructor() {
    companion object {
        fun isForced(extras: Bundle) = extras.getBoolean(ContentResolver.SYNC_EXTRAS_MANUAL, false)

        fun <E : Base> index(items: Collection<E>) = LongSparseArray<E>().apply {
            for (item in items) put(item.id, item)
        }
    }
}
